src/main/groovy/jenkins/automation/utils/GhUtils.groovy

```diff
@@ -12,51 +14,87 @@ class GhUtils {
+                      Valid keys:
+ * <br/>&#x2014;  <code>ghProject</code> The GitHubg project name, e.g. username/reponame
+ * <br/>&#x2014;  <code>ghHostname</code> The GitHub hostname, i.e. github.org.tld,
+ *                or github.com
+ * <br/>&#x2014;  <code>ghAuthId</code> the Jenkins credential ID for GitHub authentication
+ *                see <a href="https://github.com/jenkinsci/ghprb-plugin/blob/master/README.md#credentials">the plugin docs</a>
+ * <br/>&#x2014;  <code>ghPermitAll</code>  optional Whether or not to allow all
+ *                contributors to build a PR. Defaults to false.
+ * <br/>&#x2014;  <code>ghPrHooks</code>  optional Whether or not to install webhooks
+ *                in the remote repo. Defaults to true.
+ * <br/>&#x2014;  <code>ghPrCron</code>  optional For use when ghPrHooks is false.
+ *                Uses standard cron syntax.
+ * <br/>&#x2014;  <code>ghPrOrgList</code>  optional A string of organizations to whitelist for
+ *                PRs. Defaults to 'jenkins'.
+ * <br/>&#x2014;  <code>ghPrStatusContext</code>  optional A context for this job's tests as it
+ *                will appear on PR issues in GH's UI. Defaults to 'Tests from GitHub PR Builder.'
+ * <br/>&#x2014;  <code>ghPrResultMessage</code>  optional A <code>Map</code> whose keys are the build
+ *                result (SUCCESS, FAILURE, or ERROR) and value is the message posted to GH as a result.
  */
-    static void ghPrWatcher(context,
-                            String ghProject,
-                            String ghHostname,
-                            String ghAuthId,
-                            Boolean ghPermitAll = false,
-                            Boolean ghPrHooks = true,
-                            String ghPrCron = '',
-                            String ghPrOrgsList = 'jenkins'
-    ) {
+    static void ghPrWatcher(context, Map ghPrConfig) {
+        def defaults = ghPrConfigDefaults
         context.with {
             scm {
                 git {
                     remote {
-                        github(ghProject, 'https', ghHostname)
+                        github(
+                            ghPrConfig.get('ghProject'),
+                            'https',
+                            ghPrConfig.get('ghHostname')
+                        )
                         refspec('+refs/pull/*:refs/remotes/origin/pr/*')
                     }
                     branch('${sha1}')
                 }
             }
             configure { node ->
                 node / 'triggers' / 'org.jenkinsci.plugins.ghprb.GhprbTrigger'( plugin: 'ghprb') {
-                    if (ghPrCron) {
-                        spec(ghPrCron)
-                        cron(ghPrCron)
-                    }
-                    else {
-                        spec()
-                        cron()
-                    }
+                    // Avoid 'null' in these blocks
+                    spec(ghPrConfig['ghPrCron'] ?: '')
```

Comment:
This is a bit baffling, but if I try to look up the value via `get` for this particular node, then the rest run bombs out with the following:

```
No signature of method: groovy.util.NodeBuilder.spec() is applicable for argument types: (java.lang.String, java.lang.String) values: [*/2 * * * *, ]
```
I'm not 100% sure why that's happening since it works with this style look up. The Elvis operator is in there because it's cosmetically bad to leave 'null' in the config, it creates unnecessary diffs, I'm not 100% sure how XML handles the string 'null', and it's not valid cron spec.

Reply:
what about `ghPrConfig.get('ghPrCron', '')` instead?

Reply:
I tried that and preferred it, but it was throwing the above error. Let me know if you get that working in testing.

Reply:
@champain I just tried it locally, no problems. The error you included above makes it look like there was something else going on, possibly. Can you try again and see if it works?